src/bootstrap.jl

```diff
@@ -20,32 +22,25 @@ replicates: 1000
 ```
 """
 function bootweights(design::SurveyDesign; replicates=4000, rng=MersenneTwister(1234))
-    H = length(unique(design.data[!, design.strata]))
     stratified = groupby(design.data, design.strata)
-    function replicate(stratified, H)
-        for h in 1:H
-            substrata = DataFrame(stratified[h])
-            psus = unique(substrata[!, design.cluster])
-            if length(psus) <= 1
-                stratified[h].whij .= 0 # hasn't been tested yet. 
-            end
-            nh = length(psus)
-            randinds = rand(rng, 1:(nh), (nh-1)) # Main bootstrap algo. Draw nh-1 out of nh, with replacement.  
-            rh = [(count(==(i), randinds)) for i in 1:nh] # main bootstrap algo. 
-            gdf = groupby(substrata, design.cluster)
-            for i in 1:nh
-                gdf[i].whij = repeat([rh[i]], nrow(gdf[i])) .* gdf[i][!,design.weights] .* (nh / (nh - 1))
```

Comment:
@ayushpatnaikgit can you explain in a post how you achieved this speedup? I think the issue was the gdf[i], which u solved using manual counters, sorting substrata by cluster. Also there is a `push!`, which is not the fastest operation in Julia?